datafusion/common/src/scalar.rs

```diff
@@ -2072,53 +1188,48 @@ impl ScalarValue {
         }
     }
 
+    /// Wrapping addition of `ScalarValue`
+    ///
+    /// NB: operating on `ScalarValue` directly is not efficient, performance sensitive code
+    /// should instead make use of vectorized array kernels
     pub fn add<T: Borrow<ScalarValue>>(&self, other: T) -> Result<ScalarValue> {
```

Comment:
The performance of this is likely worse, however, following #7358 the only use of this logic is within the range analysis framework, the median accumulator, and Range window functions, none of which I would consider performance critical